src/foam/nanos/crunch/extra/ImplyReviewedAction.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.extra',
+  name: 'ImplyReviewedAction',
+  implements: ['foam.nanos.ruler.RuleAction'],
+
+  javaImports: [
+    'foam.core.ContextAgent',
+    'foam.core.X',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.CrunchService',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'java.util.List',
+    'static foam.mlang.MLang.*'
+  ],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+      agency.submit(x, new ContextAgent() {
+        @Override
+        public void execute(X x) {
+          // In case this rule action is misconfigured
+          if ( ! (obj instanceof UserCapabilityJunction) ) {
+            throw new RuntimeException(
+              "ImplyReviewedAction can only be triggered on UCJ objects");
+          }
+
+          // Context requirements
+          var capabilityDAO = (DAO) x.get("capabilityDAO");
+
+          var cap = (Capability) capabilityDAO.find(
+            ((UserCapabilityJunction) obj).getTargetId());
+
+          // Find the review wizard corresponding to this capability.
+          //   A list is used in case there is more than one.
+          List reviewWizards = ((ArraySink) capabilityDAO.where(AND(
+            INSTANCE_OF(ReviewWizard.class),
+            EQ(ReviewWizard.CAPABILITY_TO_REVIEW, cap.getId())
+          )).select(new ArraySink())).getArray();
+          for ( Object o : reviewWizards ) {
+            var reviewWizard = (ReviewWizard) o;
+            grantReview(x, reviewWizard);
```

Comment:
Is the predicate of this rule that this capability is Granted?

Reply:
The predicate can be anything. (on NP-side it's the condition of having SO privilages prior to completing onboarding)